src/game-logic/reducers/minePlot.js

```diff
@@ -33,29 +33,28 @@ export const minePlot = (state, x, y) => {
   const spawnedResources = ResourceFactory.instance().generateResources(
     shovelLevel
   )
-  let spawnedOre = null
+  const [spawnedResource] = spawnedResources
   let daysUntilClear = chooseRandom(daysUntilClearPeriods)
 
-  if (spawnedResources.length) {
-    // even when multiple resources are spawned, the first one is ok to use
-    // for all subsequent logic
-    spawnedOre = spawnedResources[0]
+  if (spawnedResource) {
+    const spawnChances = spawnedResources.map(({ spawnChance }) => spawnChance)
+    const minSpawnChance = Math.min(...spawnChances)
 
-    // if ore was spawned, add up to 10 days to the time to clear
-    // at random, based loosely on the spawnChance meant to make
-    // rarer ores take longer to cooldown
-    daysUntilClear += Math.round(random() * (1 - spawnedOre.spawnChance) * 10)
+    // if a resource was spawned, add up to 10 days to the time to clear at
+    // random, based loosely on the minimum spawnChance meant to make rarer
+    // resources take longer to cooldown
+    daysUntilClear += Math.round(random() * (1 - minSpawnChance) * 10)
+  }
 
-    for (let resource of spawnedResources) {
-      state = addItemToInventory(state, resource)
-    }
+  for (let resource of spawnedResources) {
+    state = addItemToInventory(state, resource)
   }
 
   state = modifyFieldPlotAt(state, x, y, () => {
     return {
       isShoveled: true,
       daysUntilClear,
-      oreId: spawnedOre ? spawnedOre.id : null,
+      oreId: spawnedResource?.id ?? null,
```

Comment:
won't the `spawnedResource?` bit already return null if it doesn't exist? i think the `?? null` might be unnecessary here

Reply:
`spawnedResource` could potentially be `undefined` here. If that's the case, `spawnedResource?.id` would be `undefined`, not `null`:

```js
const value = undefined
console.log(value?.property) // undefined
```

Setting `oreId` to `undefined ` probably wouldn't break anything, but I have a stylistic preference to use `null` when explicitly setting a non-truthy value like this (it implies that the value was set intentionally, to me).